pkg/data/management/machinedriver_data.go

```diff
@@ -104,14 +104,14 @@ func addMachineDrivers(management *config.ManagementContext) error {
 	}
 	harvesterEnabled := features.GetFeatureByName(HarvesterDriver).Enabled()
 	// make sure the version number is consistent with the one at Line 40 of package/Dockerfile
-	harvesterDriverVersion := "v0.6.9"
+	harvesterDriverVersion := "v0.7.0"
 	harvesterDriverURL := fmt.Sprintf("https://github.com/harvester/docker-machine-driver-harvester/releases/download/%s/docker-machine-driver-harvester-%s.tar.gz", harvesterDriverVersion, runtime.GOARCH)
-	harvesterDriverCheckSum := "76e1aeb778709e7350ce51f5252e87943ddcc914ad65ac487baa505c93b8e9f7"
+	harvesterDriverCheckSum := "ee28240ca31f44f4ee003040da1d92e92a2487dbd96a387c8fe647c605b27d60"
 	if runtime.GOARCH == "arm64" {
 		//overwrite arm driver version here
-		harvesterDriverCheckSum = "63727eb1832b2dec67ca512a9f534b9bf80da4938627b00bd29cbb66af5f0794"
+		harvesterDriverCheckSum = "cc15765e287298a8e2d2744367fada1838cc80a5851ce78297138230beaf4fe6"
```

Comment:
The checksum looks different.

```
> sha256sum docker-machine-driver-harvester-amd64.tar.gz docker-machine-driver-harvester-arm64.tar.gz
7d5817d521407b33aa163b50c9c33fe562bb56fffd08c862894781c7720d9ecb  docker-machine-driver-harvester-amd64.tar.gz
14aca2912dcc7949393b709fcba3e5b994e5103d3177607138ccb9986fb450bd  docker-machine-driver-harvester-arm64.tar.gz
```

Reply:
The checksums that were hardcoded previous to my commit matched the unpacked executable, not the archive. Therefore I put the checksums of the new executable files there.

Reply:
Hi @m-ildefons, I think previous checksums also match to archive.

```
> wget https://github.com/harvester/docker-machine-driver-harvester/releases/download/v0.6.9/docker-machine-driver-harvester-amd64.tar.gz

> wget https://github.com/harvester/docker-machine-driver-harvester/releases/download/v0.6.9/docker-machine-driver-harvester-arm64.tar.gz

> sha256sum docker-machine-driver-harvester-amd64.tar.gz docker-machine-driver-harvester-arm64.tar.gz
76e1aeb778709e7350ce51f5252e87943ddcc914ad65ac487baa505c93b8e9f7  docker-machine-driver-harvester-amd64.tar.gz
63727eb1832b2dec67ca512a9f534b9bf80da4938627b00bd29cbb66af5f0794  docker-machine-driver-harvester-arm64.tar.gz
```

If we check source code, I think it also use checksum for downloaded file, not unpacked executable.

1. Rancher uses url and checksum for docker machine.
https://github.com/rancher/rancher/blob/2e82262225844f52c3ef6d8a8f51b870e220ec91/pkg/controllers/capr/machineprovision/args.go#L86-L127

2. Entrypoint uses url and checksum for `download_driver.sh`.
https://github.com/rancher/machine/blob/71e1d1b7dc753e21329635e09f60b79792bc4a6a/package/entrypoint.sh#L33

3. `download_driver.sh` downloads url and compare checksum.
https://github.com/rancher/machine/blob/71e1d1b7dc753e21329635e09f60b79792bc4a6a/package/download_driver.sh#L15-L22

Reply:
Thanks!
This was indeed a mistake by me, because the sha256 checksum of the tar archive from releases.rancher.com differs from the one from github.com for the v0.6.7 release on `main` branch. However the checksums for the executable file are quite similar, so I assumed the checksums were of the executables, when I made the PR against the `main` branch.
It's fixed now :+1: